static/src/javascripts/bootstraps/enhanced/notifications.js

```diff
@@ -43,37 +48,69 @@ define([
+                $follow.html(src);
+                bean.one($follow[0], 'click', '.js-notifications__button', handler);
+            });
+        },
+
+        showExplainer: function() {
+            var src = template(explainer,{
+                closeIcon : svgs('closeCentralIcon'),
+                imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'live-notifications-explainer-svg']),
+                imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'live-notifications-explainer-svg'])
+            });
+
             fastdom.write(function () {
-                $('.js-notifications').prepend(src);
-                bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+                var $notifications = $('.js-live-notifications');
+                $notifications.append(src);
+               // bean.one($notifications[0], 'click', '.js-live-notifications__item__close', function() {
+                bean.one($('.js-live-notifications__item__close')[0], 'click', function() {
+                    fastdom.write(function() {
+                        userPrefs.set(explainerDismissed, true);
+                        $('.js-live-notifications-explainer').remove();
+                    });
+                });
             });
         },
 
-        setSubscriptionStatus: function (subscribed) {
-            var subscribeButton = $('.js-notifications-subscribe-link'),
-                subscribeLink = $('.notifications-subscribe-link--follow'),
-                handler = subscribed ? modules.unSubscribeHandler : modules.subscribeHandler;
-            if (subscribed) {
-                subscribeLink.addClass('notifications-subscribe-link--has-subscriptions');
-            }
+        closeDisplayMessage: function(){
+            $('.js-live-notifications-denied').remove();
+            bean.one($('.js-live-notifications')[0], 'click', '.js-notifications__button', modules.subscribeHandler);
+        },
 
-            subscribeButton[0].textContent = subscribed ?  'Unfollow' : 'Follow story';
-            bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+        notificationsDeniedMessage: function() {
+            var src = template(permissionsTemplate,{closeIcon : svgs('closeCentralIcon')});
+            fastdom.write(function () {
+                var blocked = $('.js-notifications-blocked');
+                blocked.prepend(src);
+                bean.one(blocked[0], 'click', '.js-live-notifications-denied__item__close', modules.closeDisplayMessage);
+            });
         },
 
         subscribeHandler: function () {
-            modules.subscribe().then(modules.follow);
+            modules.subscribe().then(modules.follow)
+                .catch( function () {
+                    if (Notification.permission === 'denied') {
+                        modules.notificationsDeniedMessage();
+                    }
+                });
```

Comment:
We should be throwing errors we don't care about, otherwise things will silently fail. It's not safe to presume any error that arrives here will be permissions related.